arcjet/index.ts

```diff
@@ -523,6 +568,163 @@ export function slidingWindow<
+  }
+
+  if (entity === "CREDIT_CARD_NUMBER") {
+    return { tag: "credit-card-number" as const };
+  }
+
+  return {
+    tag: "custom" as const,
+    val: entity,
+  };
+}
+
+function analyzeEntitiesToString(entity: analyze.SensitiveInfoEntity): string {
+  if (entity.tag === "email") {
+    return "EMAIL";
+  }
+
+  if (entity.tag === "ip-address") {
+    return "IP_ADDRESS";
+  }
+
+  if (entity.tag === "credit-card-number") {
+    return "CREDIT_CARD_NUMBER";
+  }
+
+  if (entity.tag === "phone-number") {
+    return "PHONE_NUMBER";
+  }
+
+  return entity.val;
+}
+
+function convertAnalyzeDetectedEntity(
+  detectedEntities: analyze.DetectedEntity[],
+): IdentifiedEntity[] {
+  return detectedEntities.map((detectedEntity) => {
+    return {
+      ...detectedEntity,
+      identifiedType: analyzeEntitiesToString(detectedEntity.identifiedType),
+    };
+  });
+}
+
+export function sensitiveInfo<
+  const Detect extends DetectEntities<CustomEntities>,
+  const CustomEntities extends string,
+>(
+  options: SensitiveInfoOptions<Detect, CustomEntities>,
+  ...additionalOptions: SensitiveInfoOptions<Detect, CustomEntities>[]
+): Primitive<{}> {
+  const rules: ArcjetSensitiveInfoRule<{}>[] = [];
+
+  // Always create at least one SENSITIVE_INFO rule
+  for (const opt of [options, ...additionalOptions]) {
+    const mode = opt.mode === "LIVE" ? "LIVE" : "DRY_RUN";
+    rules.push({
+      type: "SENSITIVE_INFO",
+      priority: Priority.SensitiveInfo,
+      mode,
+      allow: options.allow || [],
+      deny: options.deny || [],
+
+      validate(
+        context: ArcjetContext,
+        details: ArcjetRequestDetails,
+      ): asserts details is ArcjetRequestDetails {},
+
+      async protect(
+        context: ArcjetContext,
+        details: ArcjetRequestDetails,
+      ): Promise<ArcjetRuleResult> {
+        const body = await context.getBody();
+        if (body === undefined) {
+          return new ArcjetRuleResult({
+            ttl: 0,
+            state: "NOT_RUN",
+            conclusion: "ERROR",
+            reason: new ArcjetErrorReason(
+              "Couldn't read the body of the request to perform sensitive info identification.",
+            ),
+          });
+        }
+
+        let convertedDetect = undefined;
+        if (options.detect !== undefined) {
+          const detect = options.detect;
+          convertedDetect = (tokens: string[]) => {
+            return detect(tokens).map((e) => e && protocolEntitiesToAnalyze(e));
+          };
+        }
+
+        let entitiesTag: "allow" | "deny" = "allow";
+        let entitiesVal =
+          options?.allow
+            ?.filter((e) => e !== undefined)
+            .map(protocolEntitiesToAnalyze) || [];
+
+        if (options?.deny && options?.deny.length > 0) {
+          entitiesTag = "deny";
+          entitiesVal = options.deny
+            .filter((e) => e !== undefined)
+            .map(protocolEntitiesToAnalyze);
+        }
```

Comment:
I noticed the undefined checks near the top but the optional chaining caused me concern and I found that you were using `option` instead of `opt` which meant you were always using the first options for any given rule. Perhaps you need to add tests for configuring multiple rules with one constructor.

That being said. I really dislike the multiple-rules-in-one-constructor API and I think we should discard it for the beta.

Reply:
> That being said. I really dislike the multiple-rules-in-one-constructor API and I think we should discard it for the beta.

@blaine-arcjet Could you add a note about that to #928 please? I can't picture what you mean exactly, so it'd be good to have an example (current, and proposed change) 😊

Reply:
Added a new issue #1397

Also referenced the issue in #928